errai-codegen/src/main/java/org/jboss/errai/codegen/meta/HasAnnotations.java

```diff
@@ -53,4 +55,21 @@ default boolean unsafeIsAnnotationPresent(final Class<? extends Annotation> anno
             .findFirst()
             .orElse(null);
   }
+
+  default Optional<MetaAnnotation> getAnnotation(final Class<? extends Annotation> annotationClass) {
+    return Optional.ofNullable(unsafeGetAnnotation(annotationClass)).map(RuntimeMetaAnnotation::new);
+  }
+
+  @SuppressWarnings("unchecked")
```

Comment:
Why did this change from `boolean` to `Boolean`?

Reply:
I tend to use less primitives as possible.. I guess it's more like an old habit than anything. Since it's new code, I figured it was okay to change it. if you see any reason not to, I can change it back without problems.

Reply:
It's fine, was just curious.

Reply:
@tiagobento Just a remark - primitive types are better in client side code, because GWT compiler maps them to native js types if possible. Though it doesn't matter in this particular case.

Reply:
@slavap Thanks for mentioning that. I think since GWT 2.8.0 Booleans are not boxed anymore. See under _JDK emulation_ section: http://www.gwtproject.org/release-notes.html#Release_Notes_2_8_0_BETA1. Anyway, primitives do perform better inside the JVM, so it might be the case I rethink my preferences :P